### PR TITLE
GAWB-408: swaggerize missing endpoints (rawls/orch)

### PR DESCRIPTION
**Pull request by davidangb**
part of GAWB-408 but does not satisfy that ticket fully.

you may want to read the individual commits. See also https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/resources/swagger/README.md if you haven't read that yet.

This PR adds to swagger:
* endpoints that existed in orch but were undocumented
* endpoints that were passed through to rawls but were undocumented

I have not yet attempted to fix:
* endpoints that were passed through to agora but were undocumented
* un-commenting the billing endpoints
* verifying content for pre-existing endpoints

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -1352,36 +1458,6 @@ paths:
         500:
           description: Internal Server Error
 
-  /api/workspaces/{workspaceNamespace}/{workspaceName}/methodconfigs/{configNamespace}/{configName}/validate:
```
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml:1355]: "methodconfigs" in the path was wrong; it should be "method_configs". This rename required the entire block to be moved up in the file, to maintain alphabetization.

File: src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -1481,18 +1621,6 @@ paths:
           description: Submission or Workflow not found
         500:
           description: Internal Error
-  /api/submissions/queueStatus:
```
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml:1484]: due to previous copy-paste error, this queueStatus endpoint was in the wrong place. I relocated it to maintain alphabetization and updated its contents.

File: src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -1888,6 +2056,48 @@ definitions:
...
+      backend:
+        type: string
+        description: type of backend executing the call
+      stdout:
+        type: string
+        description: location of stdout
+      stderr:
+        type: string
+        description: location of stderr
+
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml:2107]: I also see in the call meta data json:

```
"shardIndex": -1,
"executionEvents": [{
        "description": "cromwell poll interval",
        "startTime": "2016-10-20T19:43:31.000Z",
        "endTime": "2016-10-20T19:43:49.291Z"
      },...]
```

**Comment by rushtong** [comment]
One minor comment inline.

I wasn't able to make work the following endpoints:

```
PATCH /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}/{entityName}
POST /api/workspaces/{workspaceNamespace}/{workspaceName}/submissions/validate
GET /cookie-authed/download/b/{bucket}/o/{object}
```

But that could be me just not knowing what magic words to speak. 
Otherwise , 👍

File: src/main/resources/swagger/api-docs.yaml (hunk 3 above)
Comment (davidangb) [reply src/main/resources/swagger/api-docs.yaml:2107]: good catch on this! The executionEvents was added just recently and I missed it (broadinstitute/rawls#567). I added shardIndex too, and will create a PR in rawls to update it there.

**Comment by davidangb** [comment]
I did test

```
PATCH /api/workspaces/{workspaceNamespace}/{workspaceName}/entities/{entityType}/{entityName}
POST /api/workspaces/{workspaceNamespace}/{workspaceName}/submissions/validate
```

and saw both of those working - the syntax is a bit wonky.

```
GET /cookie-authed/download/b/{bucket}/o/{object}
```

won't work in swagger due to a fundamental limitation - can't download a file via ajax. I tried to cover this in the description/Implementation Notes ... did you try that approach?

**Comment by rushtong** [comment]
I tried the approach described in the download description - couldn't figure out what the right url construction was supposed to be. I'll take another stab at working out the url components.

**Comment by davidangb** [comment]
For the cookie-authed download, try:
bucket: cancer-exome-pipeline-demo-data
object: HCC1143_BL.100_gene_250bp_pad.bai
which will lead to a url like https://local.broadinstitute.org:10443/cookie-authed/download/b/cancer-exome-pipeline-demo-data/o/HCC1143_BL.100_gene_250bp_pad.bai.

You do have to have previously logged in to FC to have the FCtoken tookie in your browser.

File: src/main/resources/swagger/api-docs.yaml [hunk 4]
```diff
@@ -1543,13 +1671,47 @@ paths:
...
+      responses:
+        200:
+          description: Success
+          schema:
+            type: file
+        404:
+          description: Not Found
+        500:
+          description: Internal Error
+      security: []
```
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml:1706]: ~~The swagger UI doesn't show you the Authorization Bearer in the curl command. I am assuming this is the reason. I added googleoauth back in and now that does show up in the curl command.~~

My bad - cookie authed doesn't need the header

**Comment by rushtong** [comment]
👍  Testing required having the UI running which generated the right cookie for the cookie-authed endpoint to work.

**Comment by rushtong** [comment]
👍

**Pull request merged**